### PR TITLE
Login rework: Prologue animations

### PR DESCRIPTION
**Pull request by hypest**
Introduces Lottie animations to the prologue screen. 4 promo screens are included in a ViewPager and auto-cycle. Manual swiping is also enabled.

The final visual touches, messages and final texts will be done in a future PR.

To test:
* With the app logged out or data-cleaned, open the app
* Notice the animations playing and the ViewPager auto-advancing after a few loops of each animation is performed

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/accounts/login/LoginPrologueAnimationFragment.java [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle
+            savedInstanceState) {
+        ViewGroup rootView = (ViewGroup) inflater.inflate(R.layout.login_intro_template_view, container, false);
+
+        TextView promoText = (TextView) rootView.findViewById(R.id.promo_text);
+        promoText.setText(mPromoText);
+
+        mLottieAnimationView = (LottieAnimationView) rootView.findViewById(R.id.animation_view);
+        mLottieAnimationView.setAnimation(mAnimationFilename, LottieAnimationView.CacheStrategy.Weak);
+//        mLottieAnimationView.setImageAssetsFolder("login_anims");
```
Comment (nbradbury) [inline WordPress/src/main/java/org/wordpress/android/ui/accounts/login/LoginPrologueAnimationFragment.java]: Should this line be removed?

**Comment by nbradbury** [comment]
This is really nice! Looks like it needs some work in landscape, though.

device-2017-06-06-123525

File: WordPress/src/main/java/org/wordpress/android/ui/accounts/login/LoginPrologueAnimationFragment.java [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+            @Override
+            public void onAnimationCancel(Animator animation) {
+                mCount = 0;
+            }
+
+            @Override
+            public void onAnimationRepeat(Animator animation) {
+                mCount++;
+
+                if (mCount > 2) {
```
Comment (nbradbury) [inline WordPress/src/main/java/org/wordpress/android/ui/accounts/login/LoginPrologueAnimationFragment.java]: I'm wondering if `(mCount > 1)` would be better? Three loops seems a bit much.

**Comment by nbradbury** [comment]
What do you think about slowing down the automatic page transition? The transition happens very fast, and I'm thinking it might seem smoother if it was slower.

File: WordPress/src/main/java/org/wordpress/android/ui/accounts/login/LoginPrologueAnimationFragment.java (hunk 1 above)
Comment (hypest) [inline WordPress/src/main/java/org/wordpress/android/ui/accounts/login/LoginPrologueAnimationFragment.java]: That line is "needed" for the other commented out line: https://github.com/wordpress-mobile/WordPress-Android/pull/5987/files#diff-6e36ed97c1ebc7e97bd01854b74874dcR15

I'll remove both (and the corresponding images in the assets folder) and add them again in the future, when that animation gets worked on.

File: WordPress/src/main/java/org/wordpress/android/ui/accounts/login/LoginPrologueAnimationFragment.java (hunk 2 above)
Comment (hypest) [inline WordPress/src/main/java/org/wordpress/android/ui/accounts/login/LoginPrologueAnimationFragment.java]: It is kinda much if you only focus your attention on the anims. When you try to read the promo texts though (which are rather long I'd say) at least 1 of the loops is consumed, leaving 2 for watching the anim themselves. I think I'll keep the 3 loops for now but that's something we can easily adjust at the visual design pass anyway. Cool?

**Comment by hypest** [comment]
Hey @nbradbury , thanks for picking the PR up for review!

> Looks like it needs some work in landscape, though.

Definitely. I had a solution in mind but thought of leaving it for later. I will give it a go now.

> What do you think about slowing down the automatic page transition? The transition happens very fast, and I'm thinking it might seem smoother if it was slower.

I too find it too fast for my liking. The ViewPager API doesn't seem to have a configuration for that though. Then again, we already have 4 pages of anims, with each page having both long-ish text and a unique anim looping 3 times. I would say that the whole feature already takes enough time to complete and the user is still on the very first of the app, hopefully eager to Log in. Prolonging that time probably is too much. Let's leave this as is, at least until the visual touches phase, wdyt?

**Comment by hypest** [comment]
Ready for another pass @nbradbury , thanks!

**Comment by nbradbury** [comment]
Looks great! :shipit:

**Pull request merged**